### PR TITLE
feat: life events

### PR DESCRIPTION
**Pull request by djaiss**
The core of the app.😱

Many many, many features.
- Life events on the dashboard in a new tab
- Dashboard: we remember the tab position now
- Add life events on a contact profile page also

So. Summary:

- A contact has a timeline of events that happened to him.
- A timeline is composed of TimelineEvent objects.
- Each TimelineEvent can have one or more LifeEvent objects.

Why do we have a timeline event and a life event objects?

Yes, life events can be "Drove a car" or "Went to cinema". But these are single actions.

A timeline event contains one or more life events. So you can use it to represent an entire evening (I saw my friends) which contains multiple actions (restaurant, then cinema, then whatever). Also, a timeline event can span multiple days or weeks even if you so desire (like "Summer family vacation" which lasted 2 weeks).

### CODE REVIEW

File: app/Domains/Contact/ManageLifeEvents/Web/Controllers/ContactModuleTimelineEventController.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            'vault_id' => $vaultId,
+            'contact_id' => $contactId,
+            'loan_id' => $loanId,
+            'currency_id' => $request->input('currency_id'),
+            'type' => $request->input('type'),
+            'name' => $request->input('name'),
+            'description' => $request->input('description'),
+            'loaner_ids' => $loaners,
+            'loanee_ids' => $loanees,
+            'amount_lent' => $request->input('amount_lent') ? $request->input('amount_lent') * 100 : null,
```
Comment (asbiin) [inline app/Domains/Contact/ManageLifeEvents/Web/Controllers/ContactModuleTimelineEventController.php]: All currency are not working like that, this needs more backend

File: app/Models/TimelineEvent.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+                $firstEvent = $lifeEvents->sortBy('happened_at')->first();
+                $lastEvent = $lifeEvents->sortByDesc('happened_at')->first();
+
+                if ($firstEvent->id === $lastEvent->id) {
+                    return DateHelper::format($firstEvent->happened_at, Auth::user());
+                }
+
+                return DateHelper::format($firstEvent->happened_at, Auth::user()).
+                    ' - '.
```
Comment (asbiin) [inline app/Models/TimelineEvent.php]:
```suggestion
                    ' — '.
```

File: database/migrations/2023_02_01_021117_add_default_dashboard_tab_to_vaults.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('vaults', function (Blueprint $table) {
+            //
+        });
+    }
+};
```
Comment (asbiin) [inline database/migrations/2023_02_01_021117_add_default_dashboard_tab_to_vaults.php]: You can create a breaking change (like in the previous migration file), **OR** create a new migration file for 1 column, but not both. Please add this migration in the original `vaults` table creation migration.

File: resources/js/Shared/Modules/LifeEvent.vue [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+      response.data.data.timeline_events.forEach((entry) => {
+        localTimelines.value.push(entry);
+      });
+      paginator.value = response.data.paginator;
+    })
+    .catch(() => {});
+};
+
+const destroy = (timelineEvent) => {
+  if (confirm('Are you sure? This will delete the event permanently.')) {
```
Comment (asbiin) [inline resources/js/Shared/Modules/LifeEvent.vue:53]: Can you use the `ConfirmationModal` module instead?

**Review by asbiin — CHANGES_REQUESTED** [review]
See my comments:

File: app/Domains/Contact/ManageLifeEvents/Web/Controllers/ContactModuleTimelineEventController.php (hunk 1 above)
Comment (djaiss) [reply app/Domains/Contact/ManageLifeEvents/Web/Controllers/ContactModuleTimelineEventController.php]: Oh shit it's a mistake. I forgot to delete this method.

File: database/migrations/2023_02_01_021117_add_default_dashboard_tab_to_vaults.php (hunk 3 above)
Comment (djaiss) [reply database/migrations/2023_02_01_021117_add_default_dashboard_tab_to_vaults.php]: You are right. Deleting.

**Pull request merged**